### PR TITLE
Orderer v3: Remove system channel usage from integration tests: raft no.2

### PR DESCRIPTION
**Pull request by tock-ibm**
Signed-off-by: Yoav Tock <tock@il.ibm.com>
Change-Id: Ie801bbf66b5ae320fbb1f8a099f1cc4b2070ec94

#### Type of change
- Improvement (improvement to code, performance, etc)

#### Description

Orderer v3: Remove system channel usage from integration tests: raft no.2

#### Related issues
Epic: https://github.com/hyperledger/fabric/issues/3511
Issue: https://github.com/hyperledger/fabric/issues/3515

### CODE REVIEW

File: integration/raft/config_test.go [hunk 1]
```diff
@@ -510,22 +510,25 @@ var _ = Describe("EndToEnd reconfiguration and onboarding", func() {
 				Eventually(ordererProc.Ready(), network.EventuallyTimeout).Should(BeClosed())
 			}
 
+			By("Creating a new channel, joining all orderers")
+			channelparticipation.JoinOrderersAppChannelCluster(network, "testchannel", network.Orderers...)
```
Comment (Param-S) [inline integration/raft/config_test.go:514]: Can we have the channel name in a variable & reference same variable in all other statements.

File: integration/raft/config_test.go [hunk 2]
```diff
@@ -814,29 +832,36 @@ var _ = Describe("EndToEnd reconfiguration and onboarding", func() {
...
+				"testchannel": 8,
 			}, orderers, peer, network)
 
 			By("Ensuring orderer4 doesn't serve testchannel2 and testchannel3")
 			env = CreateBroadcastEnvelope(network, peer, "testchannel2", []byte("hello"))
 			resp, err = ordererclient.Broadcast(network, o4, env)
 			Expect(err).NotTo(HaveOccurred())
-			Expect(resp.Status).To(Equal(common.Status_SERVICE_UNAVAILABLE))
+			Expect(resp.Status).To(Equal(common.Status_BAD_REQUEST))
+			Expect(orderer4Runner.Err()).To(gbytes.Say("channel creation request not allowed because the orderer system channel is not defined"))
```
Comment (Param-S) [inline integration/raft/config_test.go:858]: This error message is misleading. Are we planning to update this message later?

File: integration/raft/config_test.go (hunk 1 above)
Comment (tock-ibm) [reply integration/raft/config_test.go:514]: I actually prefer it this way, having an explicit "testchannel" helps in readability. Moreover, this is the style in 90% of the integration tests so it is a bit late in the game to try and impose a different style.

File: integration/raft/config_test.go (hunk 2 above)
Comment (tock-ibm) [reply integration/raft/config_test.go:858]: Good point. Yes. When system channel support will be removed from the registrar we'll need to change this response.

File: integration/raft/config_test.go [hunk 3]
```diff
@@ -1895,3 +2103,49 @@ func updateOrdererMSPAndConsensusMetadata(network *nwo.Network, peer *nwo.Peer,
...
+
+	genesisBlock.Header.Number = 0
+	genesisBlock.Header.PreviousHash = nil
+
+	var metadataContents [][]byte
+	for i := 0; i < len(common.BlockMetadataIndex_name); i++ {
+		metadataContents = append(metadataContents, []byte{})
+	}
+	genesisBlock.Metadata = &common.BlockMetadata{Metadata: metadataContents}
+	//genesisBlock.Metadata.Metadata[common.BlockMetadataIndex_LAST_CONFIG] = protoutil.MarshalOrPanic(&common.Metadata{
```
Comment (yacovm) [inline integration/raft/config_test.go]: let's remove commented out code

File: integration/raft/config_test.go [hunk 4]
```diff
@@ -1895,3 +2103,49 @@ func updateOrdererMSPAndConsensusMetadata(network *nwo.Network, peer *nwo.Peer,
...
+		Value: protoutil.MarshalOrPanic(&common.OrdererBlockMetadata{
+			LastConfig: &common.LastConfig{Index: 0},
+		}),
+	})
+
+	envelope, err := protoutil.GetEnvelopeFromBlock(genesisBlock.Data.Data[0])
+	Expect(err).NotTo(HaveOccurred())
+	payload, err := protoutil.UnmarshalPayload(envelope.Payload)
+	Expect(err).NotTo(HaveOccurred())
+	// configEnv := &common.ConfigEnvelope{}
```
Comment (yacovm) [inline integration/raft/config_test.go]: same

Comment (tock-ibm) [reply integration/raft/config_test.go]: done

File: integration/raft/config_test.go (hunk 3 above)
Comment (tock-ibm) [reply integration/raft/config_test.go]: done

**Review by yacovm — APPROVED** [review]

**Pull request merged**